deploy.py

```diff
@@ -13,105 +13,86 @@ def add_cli_args():
                                  "openlab-ha"],
                         help='OpenLab deployment type',
                         )
+    parser.add_argument('--action',
+                        choices=["deploy", "new-slave", "switch-role",
```

Comment:
Merge all actions into one can make cli easy to use IMO. :)

Reply:
I do not think this is more easy to use than before, personal thoughts,  `--action` make things more secluded and have to run `help` firstly and then to know how to choose an action and what actions supported, so lets see others opinions

Reply:
Before this PR, when I try to use the `deploy` script, I find that there is a lot of parameters, they must be used in some specified ways, otherwise the result is unexpected. For example,

what will happen if both `switch-role` and `new-slave` are provided? or like ``switch-role` and `--graph` are both provided? `--vars` must be used when `--graph` is provided,  but what if `switch-role` and `--vars` are provided at the same time?
 
This are different actions, but they can be used at the same time in one command, Isn't is strange? For a user of `deploy` script, Even after reading the help message, I still don't know what is the correct combinations for these parameters.  I just understand how to use it after read the source code.

BTW, doesn't reading help message is always the first step to use a cli?

Reply:
just like openstack command tool, also have many parameter, and there is not a `--action` to specify the `start`, `stop`, `reboot`, `rebuild` ... such as actions. you do not need care about these arguments if you clearly know what you want to do. for example, you want to deploy an ha deployment, you only need to specify the type. yes, I also think the new-slave and swith-role is not clear definitions. better need to change. 
there is not a `--vars` parameters. only has `--with-vars` and `-extra-vars`, actually, these two and `--graph` is from Ansible itself support, only provided here for convenient check and extension. so I don't think it is strange. one parameter only used with another parameter is also an usual usage.

I remember I have mentioned about the parameter validation, I told these is not for common users, we are developers and this is for developers, so I didn't spent time to add validation about two actions specified in the same time case... so I can improve to make you happy if I have enough time and if you really care about these. we have readme and the help message.

sure if I clearly know about what I am going to do, I don't need to ready the help messages firstly, e.g. I just want to deploy an allinone for testing. I think the README is enough.

Reply:
Again, I am not strongly against this, every one has different personal habit, if others think OK, go ahead please.

Reply:
Er, how about go first for test, as the coming deadline? And I think once we test the most part of cases, then we will write a good doc for decribing the issue we faced, if all of us think there is still a issue here at that moment, we can start refactor there. Haha.. so let's merge?

Reply:
that is not a good reason.. but.. all right, if you like. please go.